src/Handler/Search.hs

```diff
@@ -86,7 +86,12 @@ getSearchR = do
     isSimpleType _ = False
 
     lookupCurrentPage :: Handler Int
-    lookupCurrentPage = maybe 1 (max 1) . ((readMaybe . unpack) =<<) <$> lookupGetParam "page"
+    lookupCurrentPage = fmap (fromMaybe 1) <$>
+        runMaybeT $ do
```

Comment:
I think this would be slightly clearer without `MaybeT`, actually:

```haskell
do
  mpage <- lookupGetParam "page"
  return $ case mpage >>= (readMaybe . unpack) of
    Just page -> max 1 $ min maxPages page
    Nothing -> 1